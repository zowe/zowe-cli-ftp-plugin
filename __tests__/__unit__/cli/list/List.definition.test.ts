/*
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Copyright Contributors to the Zowe Project.
 *
 */

import * as ListDefinition from "../../../../src/cli/list/List.definition";

describe("List definition", () => {

    it("should match the snapshot", () => {
        ListDefinition.children.forEach((child) => {
            child.handler = child.handler.replace(/^.*[\\/]/, "");

            // Remove console color control characters from snapshot.
            if (child.positionals && child.positionals.length > 0) {
                const description = child.positionals[0].description;
                // eslint-disable-next-line no-control-regex
                child.positionals[0].description = description.replace(/\x1b\[33m/g, "").replace(/\x1b\[39m/g, "");
            }
        });
        expect(ListDefinition).toMatchSnapshot();
    });

});
