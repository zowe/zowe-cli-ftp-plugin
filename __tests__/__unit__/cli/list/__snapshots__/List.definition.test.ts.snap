// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List definition should match the snapshot 1`] = `
Object {
  "aliases": Array [
    "ls",
  ],
  "children": Array [
    Object {
      "aliases": Array [
        "j",
        "js",
      ],
      "description": "List all data sets that match a DSLEVEL pattern (see help below). ",
      "examples": Array [
        Object {
          "description": "List all jobs with names beginning beginning with \\"ibmu\\"",
          "options": "--prefix \\"ibmu*\\"",
        },
        Object {
          "description": "List Alice's jobs with names beginning beginning with \\"ibmu\\"",
          "options": "--prefix \\"ibmu*\\" --owner \\"alice\\"",
        },
      ],
      "handler": "Jobs.Handler",
      "name": "jobs",
      "options": Array [
        Object {
          "aliases": Array [],
          "description": "Specify the job name prefix of the jobs you own and want to list. You can specify a wildcard, which is usually in the form \\"JOB*\\". 
 If you don't specify this option all jobs under this owner will be listed by default.",
          "name": "prefix",
          "required": false,
          "type": "string",
        },
        Object {
          "aliases": Array [
            "o",
          ],
          "description": "Specify the owner user ID of the jobs you want to list. The owner is the individual/user who submitted the job OR the user ID assigned to the job. 
 ",
          "name": "owner",
          "required": false,
          "type": "string",
        },
        Object {
          "aliases": Array [
            "s",
          ],
          "description": "Specify the status of the jobs you want to list. The status is one of the following: INPUT, ACTIVE and OUTPUT. 
 ",
          "name": "status",
          "required": false,
          "type": "string",
        },
      ],
      "outputFormatOptions": true,
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "List jobs by prefix",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "ds",
      ],
      "description": "List all data sets that match a DSLEVEL pattern (see help below). 
The following values can be used with the --response-format-filter (--rff) argument to display more data from the data sets:volume, unit, referred, ext, used, recfm, lrecl, blksz, dsorg, and dsname.",
      "examples": Array [
        Object {
          "description": "List all data sets beginning with \\"ibmuser\\" and ending in \\"cntl\\"",
          "options": "\\"ibmuser.**.cntl\\"",
        },
      ],
      "handler": "DataSet.Handler",
      "name": "data-set",
      "outputFormatOptions": true,
      "positionals": Array [
        Object {
          "description": "The pattern or patterns to match data sets against. Also known as 'DSLEVEL', it is somewhat similar to the concept of a 'glob' (but not identical). The following special sequences can be used in the pattern:
%: Matches any single character
*: Matches any number of characters within a data set name qualifier (e.g. \\"ibmuser.j*.old\\" matches \\"ibmuser.jcl.old\\" but not \\"ibmuser.jcl.very.old\\")
**: Matches any number of characters within any number of data set name qualifiers (e.g. \\"ibmuser.**.old\\" matches both \\"ibmuser.jcl.old\\" and \\"ibmuser.jcl.very.old\\")
However, the pattern cannot begin with any of these sequences.
",
          "name": "pattern",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "List data sets matching pattern",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "members",
      ],
      "description": "List all members of the specified PDS or PDSE data set.",
      "examples": Array [
        Object {
          "description": "List all members in data set \\"ibmuser.test.cntl\\"",
          "options": "\\"ibmuser.test.cntl\\"",
        },
      ],
      "handler": "DataSetMembers.Handler",
      "name": "data-set-members",
      "outputFormatOptions": true,
      "positionals": Array [
        Object {
          "description": "The PDS or PDSE data set name.",
          "name": "dsname",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "List PDS or PDSE data set members",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "sfbj",
      ],
      "description": "Given a z/OS job JOBID, list the spool files (DDs) for a z/OS job on the JES/spool queues. The command does not pre-validate the JOBID.",
      "examples": Array [
        Object {
          "description": "List the spool files of the job with JOBID JOB00123",
          "options": "job00123",
        },
      ],
      "handler": "SpoolFilesByJobid.Handler",
      "name": "spool-files-by-jobid",
      "outputFormatOptions": true,
      "positionals": Array [
        Object {
          "description": "The z/OS JOBID of the job with the spool files you want to list. No pre-validation of the JOBID is performed.",
          "name": "jobid",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "List spool files of a z/OS job",
      "type": "command",
    },
    Object {
      "aliases": Array [
        "uss",
        "uf",
      ],
      "description": "List USS files and subdirectories in a directory. Optional file name pattern like \\"prefix*\\", \\"*suffix\\", or \\"prefix*suffix\\" can be specified at the end of directory. See EXAMPLES section.

The following values can be used with the --response-format-filter (--rff) argument to display more data from the data sets: name, size, owner, group, and permissions.",
      "examples": Array [
        Object {
          "description": "List USS files in the directory \\"/u/users/ibmuser/\\"",
          "options": "\\"/u/users/ibmuser\\"",
        },
        Object {
          "description": "List USS files with suffix of \\".txt\\" in the directory \\"/u/users/ibmuser/\\"",
          "options": "\\"/u/users/ibmuser/*.txt\\"",
        },
        Object {
          "description": "List USS files in the directory \\"/u/users/ibmuser/\\" and show only the file name",
          "options": "\\"/u/users/ibmuser/\\" --rff name",
        },
      ],
      "handler": "UssFiles.Handler",
      "name": "uss-files",
      "outputFormatOptions": true,
      "positionals": Array [
        Object {
          "description": "The USS directory to list files in, or the directory with file name pattern",
          "name": "directory",
          "required": true,
          "type": "string",
        },
      ],
      "profile": Object {
        "optional": Array [
          "zftp",
        ],
      },
      "summary": "List USS files in a directory",
      "type": "command",
    },
  ],
  "description": "List data sets, data set members, uss files, jobs, spool files",
  "name": "list",
  "passOn": Array [
    Object {
      "ignoreNodes": Array [
        Object {
          "type": "group",
        },
      ],
      "merge": true,
      "property": "options",
      "value": Array [
        Object {
          "aliases": Array [
            "H",
          ],
          "description": "The hostname or IP address of the z/OS server to connect to.",
          "group": "FTP Connection options",
          "name": "host",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "P",
          ],
          "defaultValue": 21,
          "description": "The port of the z/OS FTP server.",
          "group": "FTP Connection options",
          "name": "port",
          "type": "number",
        },
        Object {
          "aliases": Array [
            "u",
          ],
          "description": "Username for authentication on z/OS",
          "group": "FTP Connection options",
          "name": "user",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "p",
            "pass",
            "pw",
          ],
          "description": "Password to authenticate to FTP.",
          "group": "FTP Connection options",
          "name": "password",
          "type": "string",
        },
        Object {
          "defaultValue": true,
          "description": "Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990). Note: Unfortunately, this plugin's functionality only works with FTP and FTPS, not 'SFTP' which is FTP over SSH.",
          "group": "FTP Connection options",
          "name": "secure-ftp",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "ru",
          ],
          "defaultValue": null,
          "description": "Reject self-signed certificates. Only specify this if you are connecting to a secure FTP instance.",
          "group": "TLS / Secure Connection options",
          "name": "reject-unauthorized",
          "type": "boolean",
        },
        Object {
          "aliases": Array [
            "sn",
          ],
          "defaultValue": null,
          "description": "Server name for the SNI (Server Name Indication) TLS extension. Only specify if you are connecting securely",
          "group": "TLS / Secure Connection options",
          "name": "server-name",
          "type": "string",
        },
        Object {
          "aliases": Array [
            "ct",
          ],
          "defaultValue": 10000,
          "description": "How long (in milliseconds) to wait for the control connection to be established.",
          "group": "FTP Connection options",
          "name": "connection-timeout",
          "type": "number",
        },
      ],
    },
  ],
  "summary": "List data sets, data set members, uss files, jobs, spool files",
  "type": "group",
}
`;
